packages/@react-spectrum/s2/src/ComboBox.tsx

```diff
@@ -147,186 +159,222 @@ const iconStyles = style({
   }
 });
 
+const loadingWrapperStyles = style({
+  gridColumnStart: '1',
+  gridColumnEnd: '-1',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  marginY: 8
+});
+
+const progressCircleStyles = style({
+  size: {
+    size: {
+      S: 16,
+      M: 20,
+      L: 22,
+      XL: 26
+    }
+  },
+  marginStart: {
+    isInput: 'text-to-visual'
+  }
+});
+
+const emptyStateText = style({
+  height: {
+    size: {
+      S: 24,
+      M: 32,
+      L: 40,
+      XL: 48
+    }
+  },
+  font: {
+    size: {
+      S: 'ui-sm',
+      M: 'ui',
+      L: 'ui-lg',
+      XL: 'ui-xl'
+    }
+  },
+  display: 'flex',
+  alignItems: 'center',
+  paddingStart: 'edge-to-text'
+});
+
+export let listbox = style<{size: 'S' | 'M' | 'L' | 'XL'}>({
```

Comment:
Here to LOADER_ROW_HEIGHTS is from https://github.com/adobe/react-spectrum/pull/8110/files

Reply:
I'll defer to @yihuiliao for this one

Reply:
actually nvm, we'll need to keep `listboxItem` as well. in `menuitem`, we define `gridTemplateColumns: subgrid` but we can't do that with the virtualized listbox for everything to layout correctly so there are differences between the two